### PR TITLE
refactor metrics

### PR DESCRIPTION
**Pull request by shz117**
to enable different components initial only subset of metrics

### CODE REVIEW

File: utils/metrics.go [hunk 1]
```diff
@@ -892,6 +906,8 @@ var metricsDefs = map[MetricName]metricDefinition{
 	},
 }
 
+var brokerMetricsDefs = map[MetricName]metricDefinition{}
```
Comment (voyager-dw) [inline utils/metrics.go]: empty?

File: utils/metrics.go [hunk 2]
```diff
@@ -21,11 +21,21 @@ import (
 	"sync"
 )
 
+// ReporterType is the type of reporter
+type ReporterType int
+
+// List of reporter types
+const (
+	ReporterTypeDataNode ReporterType = iota
```
Comment (voyager-dw) [inline utils/metrics.go]: shall we add subscriber and controller too?

Comment (shz117) [reply utils/metrics.go]: subscriber and controller don't use this di yet.
we can easily add new types and sentinels when we migrate them to this

File: utils/metrics.go (hunk 1 above)
Comment (shz117) [reply utils/metrics.go]: yeah for now.
will add actual metrics when needed.

**Pull request merged**